src/map-view.ios.ts

```diff
@@ -275,20 +275,79 @@ class MapViewDelegateImpl extends NSObject implements GMSMapViewDelegate {
     }
 }
 
+@NativeClass()
+class MapVCDelegateImpl extends NSObject implements CLLocationManagerDelegate
+{
+    public static ObjCProtocols = [CLLocationManagerDelegate];
+
+    private _owner: WeakRef<MapView>;
+    public lm: CLLocationManager = new CLLocationManager();
+
+    public static initWithOwner(owner: WeakRef<MapView>): MapVCDelegateImpl {
+        let handler = <MapVCDelegateImpl>MapVCDelegateImpl.new();
+        handler._owner = owner;
+        return handler;
+    }
+
+    public locationManagerDidChangeAuthorizationStatus(manager: CLLocationManager, status: CLAuthorizationStatus)
+    {
+        let owner = this._owner.get();
+        switch (status)
+        {
+        case CLAuthorizationStatus.kCLAuthorizationStatusAuthorizedAlways:
+            console.log("Location AuthorizedAlways")
+            owner.myLocationEnabled = true
+            this.lm.startUpdatingLocation()
+
+        case CLAuthorizationStatus.kCLAuthorizationStatusAuthorizedWhenInUse:
+            console.log("Location AuthorizedWhenInUse")
+            owner.myLocationEnabled = true
+            this.lm.startUpdatingLocation()
+
+        case CLAuthorizationStatus.kCLAuthorizationStatusDenied:
+            console.log("Location Denied")
+            owner.myLocationEnabled = false
+            this.lm.stopUpdatingLocation()
+
+        case CLAuthorizationStatus.kCLAuthorizationStatusNotDetermined:
+            console.log("Location NotDetermined")
+            owner.myLocationEnabled = false
+            this.lm.stopUpdatingLocation()
+
+        case CLAuthorizationStatus.kCLAuthorizationStatusRestricted:
+            console.log("Location Restricted")
+            owner.myLocationEnabled = false
+            this.lm.stopUpdatingLocation()
+        }
+    }
+
+    public locationManagerDidUpdateLocations(manager: CLLocationManager, locations: NSArray<CLLocation> | CLLocation[])
+    {
+        let owner = this._owner.get();
+        console.log(locations)
+        /*if (locations.length > 0)
+        {
+            owner.gMap.camera = GMSCameraPosition.cameraWithTargetZoom(locations.coordinate, 10.0)
+            owner.settings.myLocationButton = true
+        }*/
+    }
+}
 
 export class MapView extends MapViewBase {
 
     protected _markers: Array<Marker> = new Array<Marker>();
 
-    private _delegate: MapViewDelegateImpl;
+    public _delegate: MapViewDelegateImpl;
```

Comment:
May I ask just for my curiosity why `_delegate` must be public ?

Reply:
I was working into different implementation for the delegate that need to be public .. i will change this with the next commit

Reply:
Isn't the getter available?

There are some problems coming from the update from TS 3.7 to 3.9

Reply:
For the moment i couldn't find a way to implement 2 delegates to attach them to the this.nativeView.delegate even I tried to implement multicastDelegate but it doesn't work ! maybe we should as Nathan Walker about the feasibility on NS.